### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by jobez**
adds `compute_starknet_address` into the flow of operations

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -15,10 +15,11 @@ services:
...
     environment:
-      - PRIVATE_KEY=0
-      - STARKNET_ACCOUNT_ADDRESS=0
-      - TOKEN_ADDRESS=0
-      - RPC_URL=0
+      - PRIVATE_KEY=0x0300001800000000300000180000000000030000000000003006001800006600
+      - STARKNET_ACCOUNT_ADDRESS=0x03ee9e18edc71a6df30ac3aca2e0b02a198fbce19b7480a63a0d71cbd76652e0
+      - TOKEN_ADDRESS=0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7
+      - KAKAROT_ADDRESS=0x66a5454898f533792a1b3563d808c680a1cf6dd485797fc7044d71d88c49205  
+      - RPC_URL=http://host.docker.internal:5050
```
Comment (jobez) [inline docker-compose.yml:22]: this assumes a dev setup, need to be mindful of how its set on deploy

File: backend/onchain.js [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
+    nodeUrl: rpcUrl,
 });
 
-const account = new starknet.Account(provider, address, process.env.PRIVATE_KEY);
+const account = new starknet.Account(provider, address, privateKey);
+
+const contract = new starknet.Contract(erc20Json, ethTokenAddress, provider);
 
-const contract = new starknet.Contract(erc20Json, eth_address, provider);
+export async function getStarknetEoaAddress(ethAddress) {
```
Comment (Eikix) [inline backend/onchain.js:29]: spicy take: i love JS / TS

argh 🙈 if only it wasn't so smelly underneath these beautiful abstraction

File: docker-compose.yml (hunk 1 above)
Comment (Eikix) [reply docker-compose.yml:22]: can these values be overridden a launch by the docker compose?
for instance if kakarot_address changes, or even if rpc is run on madara for instance

File: backend/onchain.js [hunk 3]
```diff
@@ -1,16 +1,41 @@
 import * as starknet from 'starknet';
 import erc20Json from './erc20_abi.json' assert { type: 'json' };
 
-const eth_address = process.env.TOKEN_ADDRESS;
-const address = process.env.STARKNET_ACCOUNT_ADDRESS;
+function getEnvVariable(name, defaultValue) {
+    const value = process.env[name];
```
Comment (ClementWalter) [inline backend/onchain.js:5]: this is

```
const value = process.env[name] || defaultValue;
if (value === undefined) {
    throw new Error()
}
return value
```

File: docker-compose.yml (hunk 1 above)
Comment (ClementWalter) [reply docker-compose.yml:22]: we may want to use env variable here instead, and https://docs.docker.com/compose/environment-variables/set-environment-variables/, and create env file for katana, madara, other networks with more private info, etc.

**Review by ClementWalter — CHANGES_REQUESTED** [review]
I think that we can just improve a bit the env handling to ease the usage on both madara, katana, and possible other target (sharingan)

**Pull request merged**